skbio/tree/_nj.py

```diff
@@ -175,10 +175,11 @@ def _compute_q(dm):
     """
     q = np.zeros(dm.shape)
     n = dm.shape[0]
-    for i in range(n):
-        for j in range(i):
-            q[i, j] = q[j, i] = \
-                ((n - 2) * dm[i, j]) - dm[i].sum() - dm[j].sum()
+    dmv = dm.to_data_frame().values
```

Comment:
@wasade, the specific slowdown is the `DataFrame.values` call. Internally, pandas will split its data into blocks when the `DataFrame` is created, and calling `.values` requires pandas to reconstruct the 2D numpy array from those blocks.